tests/baselines/reference/logicalAssignment2(target=es2015).js

```diff
@@ -0,0 +1,42 @@
+//// [logicalAssignment2.ts]
+interface A {
+    foo: {
+        bar(): {
+            baz: 0 | 1 | 42 | undefined | ''
+        }
+        baz: 0 | 1 | 42 | undefined | ''
+    }
+    baz: 0 | 1 | 42 | undefined | ''
+}
+
+declare const result: A
+declare const a: A
+declare const b: A
+declare const c: A
+
+a.baz &&= result.baz
+b.baz ||= result.baz
+c.baz ??= result.baz
+
+a.foo["baz"] &&= result.foo.baz
+b.foo["baz"] &&= result.foo.baz
+c.foo["baz"] &&= result.foo.baz
+
+a.foo.bar().baz &&= result.foo.bar().baz
+b.foo.bar().baz ||= result.foo.bar().baz
+c.foo.bar().baz ??= result.foo.bar().baz
+
+
+
+//// [logicalAssignment2.js]
+"use strict";
+var _a, _b;
+a.baz && (a.baz = result.baz);
+b.baz || (b.baz = result.baz);
+(_a = c.baz) !== null && _a !== void 0 ? _a : (c.baz = result.baz);
+a.foo["baz"] && (a.foo["baz"] = result.foo.baz);
+b.foo["baz"] && (b.foo["baz"] = result.foo.baz);
+c.foo["baz"] && (c.foo["baz"] = result.foo.baz);
+a.foo.bar().baz && (a.foo.bar().baz = result.foo.bar().baz);

```

Comment:
This transformation doesn't look right. We shouldn't see `a.foo.bar()` duplicated here as both `.foo` and `.bar()` could cause side effects. It should be something like this:

```js
(_a = a.foo.bar()).baz && (_a.baz = result.foo.bar().baz);
```